### ISSUE TITLE
Why is there no change in multiGPU training?

### ISSUE THREAD

**Issue by liuzhuang1024** [PhilJd/contiguous_pytorch_params#7]

**Comment by PhilJd** [comment]
Hi @lzmisscc, could you clarify what you mean by that? Are you referring to the function calls required, your training loss not going down, or something else?
Thanks!

**Comment by liuzhuang1024** [comment]
![image](https://user-images.githubusercontent.com/23006154/127971413-395b304d-bedd-4e9d-8d11-04f31e9be9a6.png)

I use the usage method in the readme to call the function but there is no time reduction and  run my program in a four GPUs environment.

Python 3.8.10
torch  1.6.0

**Comment by PhilJd** [comment]
Could you check the GPU usage? Maybe the job is input bound which hides the speedup? Also, how big is your model? I.e., how many paramter tensors does it contain?

**Comment by liuzhuang1024** [comment]
Model Paramter:
----------------------------------------------------------------
Total params: 1,642,048
Trainable params: 1,642,048
Non-trainable params: 0
----------------------------------------------------------------
Input size (MB): 0.73
Forward/backward pass size (MB): 1430.98
Params size (MB): 6.26
Estimated Total Size (MB): 1437.98
----------------------------------------------------------------

and the GPU usage not much change before, the input size is (96, 1000)

**Comment by PhilJd** [comment]
What's the GPU usage percentage given by `nvidia-smi`?

**Comment by liuzhuang1024** [comment]
> What's the GPU usage percentage given by `nvidia-smi`?

![image](https://user-images.githubusercontent.com/23006154/127976122-9a5d90f9-9c98-4eee-926a-d26b2696590d.png)

**Comment by PhilJd** [comment]
This looks like your job is input/CPU bound, so the GPU is idle 70% of the time anyway. To see a speed up with contiguous params, you'd need to get GPU utilization closer to 100% first.

**Comment by liuzhuang1024** [comment]
ok, thanks!👍🏻

**Comment by PhilJd** [comment]
I'll close this for now, feel free to re-open if this is still an issue when GPU usage is close to 100%.
